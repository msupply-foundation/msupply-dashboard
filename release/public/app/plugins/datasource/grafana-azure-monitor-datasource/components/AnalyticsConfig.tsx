import React, { FunctionComponent, useEffect, useMemo, useReducer, useState } from 'react';
import { SelectableValue } from '@grafana/data';
import { AzureCredentialsForm } from './AzureCredentialsForm';
import { InlineFormLabel, LegacyForms, Button, Alert } from '@grafana/ui';
const { Select, Switch } = LegacyForms;
import { AzureDataSourceSettings, AzureCredentials } from '../types';
import {
  getCredentials,
  getLogAnalyticsCredentials,
  isCredentialsComplete,
  updateLogAnalyticsCredentials,
  updateLogAnalyticsSameAs,
} from '../credentials';

export interface Props {
  options: AzureDataSourceSettings;
  updateOptions: (optionsFunc: (options: AzureDataSourceSettings) => AzureDataSourceSettings) => void;
  getSubscriptions: () => Promise<Array<SelectableValue<string>>>;
  getWorkspaces: (subscriptionId: string) => Promise<Array<SelectableValue<string>>>;
}

export const AnalyticsConfig: FunctionComponent<Props> = (props: Props) => {
  const { updateOptions, getSubscriptions, getWorkspaces } = props;
  const primaryCredentials = useMemo(() => getCredentials(props.options), [props.options]);
  const logAnalyticsCredentials = useMemo(() => getLogAnalyticsCredentials(props.options), [props.options]);

  const subscriptionId = logAnalyticsCredentials
    ? logAnalyticsCredentials.defaultSubscriptionId
    : primaryCredentials.defaultSubscriptionId;

  // Only show a section for setting LogAnalytics credentials if they were set from before
  // And the authType is supported
  const [credentialsUsed, _] = useState(!!logAnalyticsCredentials);
  const credentialsEnabled = credentialsUsed && primaryCredentials.authType === 'clientsecret';

  const hasRequiredFields =
    subscriptionId &&
    (logAnalyticsCredentials
      ? isCredentialsComplete(logAnalyticsCredentials)
      : isCredentialsComplete(primaryCredentials));

  const defaultWorkspace = props.options.jsonData.logAnalyticsDefaultWorkspace;

  const [workspaces, setWorkspaces] = useState<SelectableValue[]>([]);
  const [loadWorkspaces, onLoadWorkspaces] = useReducer((val) => val + 1, 0);
  useEffect(() => {
    if (!hasRequiredFields || !subscriptionId) {
      updateWorkspaces([]);
      return;
    }
    let canceled = false;
    getWorkspaces(subscriptionId).then((result) => {
      if (!canceled) {
        updateWorkspaces(result);
      }
    });
    return () => {
      canceled = true;
    };
    // This effect is intended to be called only once initially and on Load Workspaces click
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [loadWorkspaces, subscriptionId]);

  const updateWorkspaces = (received: Array<SelectableValue<string>>) => {
    setWorkspaces(received);
    if (!defaultWorkspace && received.length > 0) {
      // Setting the default workspace if workspaces received but no default workspace selected
      updateOptions((options) => {
        return {
          ...options,
          jsonData: {
            ...options.jsonData,
            logAnalyticsDefaultWorkspace: received[0].value,
          },
        };
      });
    } else if (defaultWorkspace) {
      const found = received.find((opt) => opt.value === defaultWorkspace);
      if (!found) {
        // Unsetting the default workspace if it isn't found among the received workspaces
        updateOptions((options) => {
          return {
            ...options,
            jsonData: {
              ...options.jsonData,
              logAnalyticsDefaultWorkspace: undefined,
            },
          };
        });
      }
    }
  };

  const [sameAsSwitched, setSameAsSwitched] = useState(false);

  const onCredentialsChange = (updatedCredentials: AzureCredentials) => {
    updateOptions((options) => updateLogAnalyticsCredentials(options, updatedCredentials));
  };

  const onLogAnalyticsSameAsChange = (event: React.SyntheticEvent<HTMLInputElement>) => {
    const sameAs = event.currentTarget.checked;
    updateOptions((options) => updateLogAnalyticsSameAs(options, sameAs));
    setSameAsSwitched(true);
  };

  const onDefaultWorkspaceChange = (selected: SelectableValue<string>) => {
    updateOptions((options) => {
      return {
        ...options,
        jsonData: {
          ...options.jsonData,
          logAnalyticsDefaultWorkspace: selected.value || '',
        },
      };
    });
  };

  const tooltipAttribute = {
    ...(!logAnalyticsCredentials && {
      tooltip: 'Workspaces are pulled from default subscription selected above.',
    }),
  };

  const showSameAsHelpMsg =
    credentialsEnabled &&
    sameAsSwitched &&
    primaryCredentials.authType === 'clientsecret' &&
    !primaryCredentials.clientSecret;

  return (
    <>
      <h3 className="page-heading">Azure Monitor Logs</h3>
      {credentialsEnabled && (
        <>
          <Switch
            label="Same details as Azure Monitor API"
            checked={!logAnalyticsCredentials}
            onChange={onLogAnalyticsSameAsChange}
            {...tooltipAttribute}
          />

          {showSameAsHelpMsg && (
            <div className="grafana-info-box m-t-2">
              <div className="alert-body">
                <p>Re-enter your Azure Monitor Client Secret to use this setting.</p>
              </div>
            </div>
          )}

          {logAnalyticsCredentials && (
            <>
              <Alert severity="info" title="Deprecated">
                Using different credentials for Azure Monitor Logs is deprecated and will be removed in a future
                version.
                <br />
                Create a different Data Source if you need to use different credentials.
              </Alert>

              <AzureCredentialsForm
                managedIdentityEnabled={false}
                credentials={logAnalyticsCredentials}
                onCredentialsChange={onCredentialsChange}
                getSubscriptions={getSubscriptions}
              />
            </>
          )}
        </>
      )}
      <div className="gf-form-group">
        <div className="gf-form-inline">
          <div className="gf-form">
            <InlineFormLabel
              className="width-12"
              tooltip="Choose the default/preferred Workspace for Azure Log Analytics queries."
            >
              Default Workspace
            </InlineFormLabel>
            <div className="width-25">
              <Select
                value={workspaces.find((opt) => opt.value === defaultWorkspace)}
                options={workspaces}
                onChange={onDefaultWorkspaceChange}
              />
            </div>
          </div>
        </div>
        <div className="gf-form-inline">
          <div className="gf-form">
            <div className="max-width-30 gf-form-inline">
              <Button
                variant="secondary"
                size="sm"
                type="button"
                onClick={onLoadWorkspaces}
                disabled={!hasRequiredFields}
              >
                Load Workspaces
              </Button>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default AnalyticsConfig;
