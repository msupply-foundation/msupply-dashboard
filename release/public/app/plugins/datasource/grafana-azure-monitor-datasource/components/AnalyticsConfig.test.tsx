import React from 'react';
import { render, screen } from '@testing-library/react';
import AnalyticsConfig, { Props } from './AnalyticsConfig';

const setup = (propsFunc?: (props: Props) => Props) => {
  let props: Props = {
    options: {
      id: 21,
      uid: 'x',
      orgId: 1,
      name: 'Azure Monitor-10-10',
      type: 'grafana-azure-monitor-datasource',
      typeName: 'Azure',
      typeLogoUrl: '',
      access: 'proxy',
      url: '',
      password: '',
      user: '',
      database: '',
      basicAuth: false,
      basicAuthUser: '',
      basicAuthPassword: '',
      withCredentials: false,
      isDefault: false,
      secureJsonFields: {
        logAnalyticsClientSecret: false,
      },
      jsonData: {
        cloudName: '',
        subscriptionId: '',
        azureLogAnalyticsSameAs: false,
        logAnalyticsDefaultWorkspace: '',
        logAnalyticsTenantId: '',
      },
      secureJsonData: {
        logAnalyticsClientSecret: '',
      },
      version: 1,
      readOnly: false,
    },
    updateOptions: jest.fn(),
    getSubscriptions: jest.fn(),
    getWorkspaces: jest.fn(),
  };

  if (propsFunc) {
    props = propsFunc(props);
  }

  return render(<AnalyticsConfig {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should disable log analytics credentials form', () => {
    const wrapper = setup((props) => ({
      ...props,
      options: {
        ...props.options,
        jsonData: {
          ...props.options.jsonData,
          azureLogAnalyticsSameAs: true,
        },
      },
    }));
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should enable azure log analytics load workspaces button', () => {
    const wrapper = setup((props) => ({
      ...props,
      options: {
        ...props.options,
        jsonData: {
          ...props.options.jsonData,
          azureLogAnalyticsSameAs: false,
          logAnalyticsDefaultWorkspace: '',
          logAnalyticsTenantId: 'e7f3f661-a933-4b3f-8176-51c4f982ec48',
          logAnalyticsClientId: '44693801-6ee6-49de-9b2d-9106972f9572',
          logAnalyticsSubscriptionId: 'e3fe4fde-ad5e-4d60-9974-e2f3562ffdf2',
          logAnalyticsClientSecret: 'cddcc020-2c94-460a-a3d0-df3147ffa792',
        },
      },
    }));
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should not render the Switch to use different creds for log analytics by default', () => {
    setup((props) => ({
      ...props,
      options: {
        ...props.options,
        jsonData: {
          ...props.options.jsonData,
          azureLogAnalyticsSameAs: undefined,
        },
      },
    }));
    expect(screen.queryByLabelText('Same details as Azure Monitor API')).not.toBeInTheDocument();
    expect(screen.queryByText('is deprecated', { exact: false })).not.toBeInTheDocument();
  });

  // Remove this test with deprecated code
  it('should not render the Switch if different creds for log analytics were set from before', () => {
    setup((props) => ({
      ...props,
      options: {
        ...props.options,
        jsonData: {
          ...props.options.jsonData,
          azureLogAnalyticsSameAs: false,
        },
      },
    }));
    expect(screen.queryByLabelText('Same details as Azure Monitor API')).toBeInTheDocument();
    expect(screen.queryByText('is deprecated', { exact: false })).toBeInTheDocument();
  });
});
