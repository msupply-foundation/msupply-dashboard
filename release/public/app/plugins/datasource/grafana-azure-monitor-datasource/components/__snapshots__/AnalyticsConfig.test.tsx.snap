// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should disable log analytics credentials form 1`] = `
<body>
  <div>
    <h3
      class="page-heading"
    >
      Azure Monitor Logs
    </h3>
    <div
      class="gf-form-group"
    >
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Default Workspace
            <div
              class="gf-form-help-icon gf-form-help-icon--right-normal"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </div>
          </label>
          <div
            class="width-25"
          >
            <div>
              <div
                class="gf-form-input gf-form-input--form-dropdown css-1daubv-SelectContainer"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="gf-form-select-box__control css-1g8ily9-Control"
                >
                  <div
                    class="gf-form-select-box__value-container css-pilx6v-ValueContainer"
                  >
                    <div
                      class="gf-form-select-box__placeholder css-ovmm9z-Placeholder"
                    >
                      Choose
                    </div>
                    <div
                      class="css-17yls17-Input"
                    >
                      <div
                        class="gf-form-select-box__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-4-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="gf-form-select-box__indicators css-1pmvkci-IndicatorsContainer"
                  >
                    <div
                      class="css-1vzus6i-Icon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <div
            class="max-width-30 gf-form-inline"
          >
            <button
              class="css-17xyotq-button"
              disabled=""
              type="button"
            >
              <span
                class="css-1mhnkuh"
              >
                Load Workspaces
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Render should enable azure log analytics load workspaces button 1`] = `
<body>
  <div>
    <h3
      class="page-heading"
    >
      Azure Monitor Logs
    </h3>
    <div
      class="gf-form-switch-container-react"
    >
      <label
        class="gf-form gf-form-switch-container "
        for="2"
      >
        <div
          class="gf-form-label   pointer"
        >
          Same details as Azure Monitor API
        </div>
        <div
          class="gf-form-switch  "
        >
          <input
            id="2"
            type="checkbox"
          />
          <span
            class="gf-form-switch__slider"
          />
        </div>
      </label>
    </div>
    <div
      aria-label="Alert info"
      class="css-7gef9a"
    >
      <div
        class="css-ed9azx"
      >
        <div
          class="css-1vzus6i-Icon"
        />
      </div>
      <div
        class="css-mqq7gx"
      >
        <div
          class="css-19c9rx9"
        >
          Deprecated
        </div>
        <div
          class="css-1r38y0g"
        >
          Using different credentials for Azure Monitor Logs is deprecated and will be removed in a future version.
          <br />
          Create a different Data Source if you need to use different credentials.
        </div>
      </div>
    </div>
    <div
      class="gf-form-group"
    >
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Directory (tenant) ID
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value="e7f3f661-a933-4b3f-8176-51c4f982ec48"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Application (client) ID
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value="44693801-6ee6-49de-9b2d-9106972f9572"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Client Secret
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Default Subscription
          </label>
          <div
            class="width-25"
          >
            <div>
              <div
                class="gf-form-input gf-form-input--form-dropdown css-1daubv-SelectContainer"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="gf-form-select-box__control css-1g8ily9-Control"
                >
                  <div
                    class="gf-form-select-box__value-container css-pilx6v-ValueContainer"
                  >
                    <div
                      class="gf-form-select-box__placeholder css-ovmm9z-Placeholder"
                    >
                      Choose
                    </div>
                    <div
                      class="css-17yls17-Input"
                    >
                      <div
                        class="gf-form-select-box__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-5-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="gf-form-select-box__indicators css-1pmvkci-IndicatorsContainer"
                  >
                    <div
                      class="css-1vzus6i-Icon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <div
            class="max-width-30 gf-form-inline"
          >
            <button
              class="css-17xyotq-button"
              disabled=""
              type="button"
            >
              <span
                class="css-1mhnkuh"
              >
                Load Subscriptions
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="gf-form-group"
    >
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Default Workspace
            <div
              class="gf-form-help-icon gf-form-help-icon--right-normal"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </div>
          </label>
          <div
            class="width-25"
          >
            <div>
              <div
                class="gf-form-input gf-form-input--form-dropdown css-1daubv-SelectContainer"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="gf-form-select-box__control css-1g8ily9-Control"
                >
                  <div
                    class="gf-form-select-box__value-container css-pilx6v-ValueContainer"
                  >
                    <div
                      class="gf-form-select-box__placeholder css-ovmm9z-Placeholder"
                    >
                      Choose
                    </div>
                    <div
                      class="css-17yls17-Input"
                    >
                      <div
                        class="gf-form-select-box__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-6-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="gf-form-select-box__indicators css-1pmvkci-IndicatorsContainer"
                  >
                    <div
                      class="css-1vzus6i-Icon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <div
            class="max-width-30 gf-form-inline"
          >
            <button
              class="css-17xyotq-button"
              disabled=""
              type="button"
            >
              <span
                class="css-1mhnkuh"
              >
                Load Workspaces
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Render should render component 1`] = `
<body>
  <div>
    <h3
      class="page-heading"
    >
      Azure Monitor Logs
    </h3>
    <div
      class="gf-form-switch-container-react"
    >
      <label
        class="gf-form gf-form-switch-container "
        for="1"
      >
        <div
          class="gf-form-label   pointer"
        >
          Same details as Azure Monitor API
        </div>
        <div
          class="gf-form-switch  "
        >
          <input
            id="1"
            type="checkbox"
          />
          <span
            class="gf-form-switch__slider"
          />
        </div>
      </label>
    </div>
    <div
      aria-label="Alert info"
      class="css-7gef9a"
    >
      <div
        class="css-ed9azx"
      >
        <div
          class="css-1vzus6i-Icon"
        />
      </div>
      <div
        class="css-mqq7gx"
      >
        <div
          class="css-19c9rx9"
        >
          Deprecated
        </div>
        <div
          class="css-1r38y0g"
        >
          Using different credentials for Azure Monitor Logs is deprecated and will be removed in a future version.
          <br />
          Create a different Data Source if you need to use different credentials.
        </div>
      </div>
    </div>
    <div
      class="gf-form-group"
    >
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Directory (tenant) ID
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Application (client) ID
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Client Secret
          </label>
          <div
            class="width-15"
          >
            <div
              style="flex-grow: 1;"
            >
              <input
                class="gf-form-input width-30"
                placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Default Subscription
          </label>
          <div
            class="width-25"
          >
            <div>
              <div
                class="gf-form-input gf-form-input--form-dropdown css-1daubv-SelectContainer"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="gf-form-select-box__control css-1g8ily9-Control"
                >
                  <div
                    class="gf-form-select-box__value-container css-pilx6v-ValueContainer"
                  >
                    <div
                      class="gf-form-select-box__placeholder css-ovmm9z-Placeholder"
                    >
                      Choose
                    </div>
                    <div
                      class="css-17yls17-Input"
                    >
                      <div
                        class="gf-form-select-box__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="gf-form-select-box__indicators css-1pmvkci-IndicatorsContainer"
                  >
                    <div
                      class="css-1vzus6i-Icon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <div
            class="max-width-30 gf-form-inline"
          >
            <button
              class="css-17xyotq-button"
              disabled=""
              type="button"
            >
              <span
                class="css-1mhnkuh"
              >
                Load Subscriptions
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="gf-form-group"
    >
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <label
            class="width-12 gf-form-label width-10"
          >
            Default Workspace
            <div
              class="gf-form-help-icon gf-form-help-icon--right-normal"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </div>
          </label>
          <div
            class="width-25"
          >
            <div>
              <div
                class="gf-form-input gf-form-input--form-dropdown css-1daubv-SelectContainer"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="gf-form-select-box__control css-1g8ily9-Control"
                >
                  <div
                    class="gf-form-select-box__value-container css-pilx6v-ValueContainer"
                  >
                    <div
                      class="gf-form-select-box__placeholder css-ovmm9z-Placeholder"
                    >
                      Choose
                    </div>
                    <div
                      class="css-17yls17-Input"
                    >
                      <div
                        class="gf-form-select-box__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-3-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="gf-form-select-box__indicators css-1pmvkci-IndicatorsContainer"
                  >
                    <div
                      class="css-1vzus6i-Icon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="gf-form-inline"
      >
        <div
          class="gf-form"
        >
          <div
            class="max-width-30 gf-form-inline"
          >
            <button
              class="css-17xyotq-button"
              disabled=""
              type="button"
            >
              <span
                class="css-1mhnkuh"
              >
                Load Workspaces
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
